aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java

```diff
@@ -320,5 +337,40 @@ private static boolean isCustomType(@NonNull Field field) {
             }
             return typeClass;
         }
+
+        private Set<SelectionSet> getModelFields(ModelSchema modelSchema, int depth) {
```

Comment:
I spent a long time trying. The challenge is in supporting custom/embedded/non-model types. The code can clean up dramatically by just using the `ModelSchema` version, but I can't figure out howto not break embedded types.

I'll leave a TODO for now, and accept temporary defeat.

Reply:
I could totally be missing something, but I'm having trouble understanding what the challenge is.  Since `modelClass` is a property of a `ModelSchema` (added in this PR),  can't you just call `modelSchema.getModelClass()` in each spot where you need a `modelClass`?  

I don't think the code related to non-model types needs to change at all - the `getNestedCustomTypeFields` will still take a `Class<?>` parameter.

Reply:
As discussed, we can come back to this later.

Reply:
As discussed, we will come back to this later.